app/jobs/in_person/enrollments_ready_for_status_check/enrollment_pipeline.rb

```diff
@@ -0,0 +1,145 @@
+module InPerson::EnrollmentsReadyForStatusCheck
+  class EnrollmentPipeline
+    # @param [InPerson::EnrollmentsReadyForStatusCheck::ErrorReporter] error_reporter
+    # @param [Regexp] email_body_pattern Pattern matching the expected email body
+    # Note: email_body_pattern must include a capture group named "enrollment_code"
+    def initialize(error_reporter:, email_body_pattern:)
+      @error_reporter = error_reporter
+      @email_body_pattern = email_body_pattern
+    end
+
+    # Process a message from USPS indicating that an in-person
+    # enrollment is ready to have its status checked.
+    #
+    # When a message can't be processed, then this function will return
+    # false to indicate that a problem occurred with the message. Otherwise it
+    # will return true. If an error occurs that can't be clearly identified
+    # as a problem with the message, then an error will be raised instead.
+    #
+    # When a message is successfully processed, then the corresponding
+    # InPersonEnrollment record will be marked as ready for a status check.
+    #
+    # @param [Aws::SQS::Types::Message] sqs_message
+    # @return [Boolean] Returns false for messages that can't be processed
+    # @raise [StandardError] Raised when an unhandled error occurs
+    def process_message(sqs_message)
+      error_extra = {
+        sqs_message_id: sqs_message.message_id,
+      }
+
+      # Unwrap SQS message to get SNS message
+      begin
+        sns_message = JSON.parse(sqs_message.body, { symbolize_names: true })
+      rescue JSON::JSONError => err
+        report_error(err, **error_extra)
+        return false
+      end
+
+      unless sns_message.is_a?(Hash) && sns_message.key?(:MessageId) && sns_message.key?(:Message)
+        report_error('SQS message body is not valid SNS payload', **error_extra)
+        return false
+      end
+
+      error_extra[:sns_message_id] = sns_message[:MessageId]
+
+      # Unwrap SNS message to get SES message
+      begin
+        ses_message = JSON.parse(sns_message[:Message], { symbolize_names: true })
+      rescue JSON::JSONError => err
+        report_error(err, **error_extra)
+        return false
+      end
+
+      unless ses_message.is_a?(Hash) && ses_message.key?(:content) && ses_message.key?(:mail)
+        report_error('SNS "Message" field is not a valid SES payload', **error_extra)
+        return false
+      end
+
+      # Add information to help with debugging
+      error_extra[:ses_aws_message_id] = ses_message.dig(:mail, :messageId)
+      error_extra[:ses_mail_timestamp] = ses_message.dig(:mail, :timestamp)
+      error_extra[:ses_mail_source] = ses_message.dig(:mail, :source)
+
+      # https://datatracker.ietf.org/doc/html/rfc5322#section-3.6.1
+      error_extra[:ses_rfc_origination_date] = ses_message.
+        dig(:mail, :commonHeaders, :date)&.tap do |date|
+          DateTime.parse(date).to_s
+        end
```

Comment:
`tap` returns the original value, not the block, I think we want a different approach here
```ruby
1.tap { |x| x + 1 }
=> 1
```

Reply:
Good catch, updated to use `then` instead. I also updated the test so that it will catch this issue.